File: LV1/햄버거 만들기.js
function solution(ingredient) {
  const stk = [];
  let cnt = 0;

  ingredient.forEach((ing) => {
    stk.push(ing);

    if (
      stk.length >= 4 &&
      stk[stk.length - 4] === 1 &&
      stk[stk.length - 3] === 2 &&
      stk[stk.length - 2] === 3 &&
      stk[stk.length - 1] === 1
    ) {
      stk.splice(stk.length - 4, 4);
      cnt++;
    }
  });

  return cnt;
}

File: LV1/햄버거 만들기.java
class Solution {

  public int solution(int[] ingredient) {
    int[] stack = new int[ingredient.length];
    int cnt = 0;
    int answer = 0;
    for (int i : ingredient) {
      stack[cnt++] = i;
      if (
        cnt >= 4 &&
        stack[cnt - 1] == 1 &&
        stack[cnt - 2] == 3 &&
        stack[cnt - 3] == 2 &&
        stack[cnt - 4] == 1
      ) {
        cnt -= 4;
        answer++;
      }
    }
    return answer;
  }
}
